Added error handling for when no data is found in `importers.dicom()`

TheSecondReal0 opened the pull request:
I used RuntimeError because no existing exception (at least that I found) fits the situation. I also didn't really want to create a custom exception since that would be more complicated to catch for users.

Atomator commented:
Looks good! That being said, could you add support/error handling if there is a directory inside the directory that's passed to  the `dir` option. Right now, if the code detects a directory inside the directory that it is reading, it will crash out. Maybe that's the behavior that we want, but I'm thinking that it should probably just ignore any directories inside a directory and just read the files.

<img width="142" alt="Screenshot 2024-03-26 at 8 10 15 AM" src="https://github.com/mines-erds/core-ct/assets/30665220/4cd142d8-7074-4dd0-89b8-0809d83653db">
<img width="430" alt="Screenshot 2024-03-26 at 8 10 22 AM" src="https://github.com/mines-erds/core-ct/assets/30665220/9d880242-cb88-4348-8098-2685ecb87720">
<img width="926" alt="Screenshot 2024-03-26 at 8 10 35 AM" src="https://github.com/mines-erds/core-ct/assets/30665220/a3237932-b2ac-4de2-922a-954f3fbaac7b">

Atomator reviewed (changes requested):
Looks good! That being said, could you add support/error handling if there is a directory inside the directory that's passed to  the `dir` option. Right now, if the code detects a directory inside the directory that it is reading, it will crash out. Maybe that's the behavior that we want, but I'm thinking that it should probably just ignore any directories inside a directory and just read the files. Ran into this problem when testing and I figured this PR would be a good place to tackle it.

<img width="142" alt="Screenshot 2024-03-26 at 8 10 15 AM" src="https://github.com/mines-erds/core-ct/assets/30665220/4cd142d8-7074-4dd0-89b8-0809d83653db">
<img width="430" alt="Screenshot 2024-03-26 at 8 10 22 AM" src="https://github.com/mines-erds/core-ct/assets/30665220/9d880242-cb88-4348-8098-2685ecb87720">
<img width="926" alt="Screenshot 2024-03-26 at 8 10 35 AM" src="https://github.com/mines-erds/core-ct/assets/30665220/a3237932-b2ac-4de2-922a-954f3fbaac7b">

TheSecondReal0 commented:
Added that check, also made it ignore non .dcm files while reading (there's an option to turn that off). I removed cases of catching and raising blank Exceptions so errors are more descriptive.

Atomator reviewed (approved):
Looks good to me! Also, I merged the branch that you approved yesterday into `main`, so just make sure to merge those changes into this branch before merging
